Controllers/OrdersManagement/ReservationsController.cs

```diff
@@ -45,53 +41,77 @@ public IEnumerable<Reservation> GetAll()
         /// <response code="404">If the item is null</response>
         // GET api/<ReservationsController>/5
         [HttpGet("{id}")]
-        public ActionResult<Reservation> Get(int id)
+        public async Task<IActionResult> Get(int id)
         {
-            var value = _randomizer.GenerateRandomData<Reservation>(id);
+            var reservation = await _context.Reservations.Where(r => r.id == id).FirstOrDefaultAsync();
+            return reservation == null ? NotFound() : Ok(reservation);
+        }
 
-            if (value == null)
-            {
-                return NotFound();
-            }
+        /// <summary>
+        /// Get awailable time slots.
+        /// </summary>
+        /// <param name="from">Start of time interval</param>
+        /// <param name="to">End of time interval</param>
+        /// <param name="interval">Splits time interval into intervals</param>
+        /// <returns></returns>
+        [HttpGet("timeslots")]
+        public async Task<IActionResult> GetTimeSlots(DateTime from, DateTime to, int interval)
+        {
+            if(from > to) return BadRequest("Time interval is not valid.");
 
-            return value;
+            var awailableTimes = await _timeSlotService.GetTimeSlots(from, to, interval);
+
+            return Ok(awailableTimes);
         }
 
         /// <summary>
-        /// Posts specific order by id to table orders
+        /// Create new resrvation
         /// </summary>
-        /// <param name="value">new created reservation</param>
+        /// <param name="reservation">new created reservation</param>
         /// <returns>one reservation by id</returns>
-        /// <response code="201">Returns the newly created item</response>
-        /// <response code="404">If the item is null</response>
+        /// <response code="200">Returns the newly created item</response>
         // POST api/<ReservationsController>
         [HttpPost]
-        public ActionResult<Reservation> Post([FromBody] Reservation value)
+        public async Task<IActionResult> Post([FromBody] CreateReservationDTO reservation)
```

Comment:
Reservation's id is a foreign key associated with Orders, so you should check if orders with that id exists as well as if reservation with such id does not exist